include/reactor-uc/macros_api.h

```diff
@@ -54,26 +54,21 @@
  */
 #define lf_is_present(trigger) (((Trigger*)(trigger))->is_present)
 
+lf_ret_t lf_schedule_with_value(Action* action, interval_t offset, const void* val);
+
 /// @private
 #define lf_schedule_with_val(action, offset, val)                                                                      \
-  do {                                                                                                                 \
+  ({                                                                                                                   \
     __typeof__(val) __val = (val);                                                                                     \
-    Action* __a = (Action*)(action);                                                                                   \
-    lf_ret_t ret = __a->schedule(__a, (offset), (const void*)&__val);                                                  \
-    if (ret == LF_FATAL) {                                                                                             \
-      LF_ERR(TRIG, "Scheduling an value, that doesn't have value!");                                                   \
-      Scheduler* sched = __a->super.parent->env->scheduler;                                                            \
-      sched->do_shutdown(sched, sched->current_tag(sched));                                                            \
-      throw("Tried to schedule a value onto an action without a type!");                                               \
-    }                                                                                                                  \
-  } while (0)
+    lf_schedule_with_value((Action*)action, offset, &__val);                                                           \
```

Comment:
It's not clear to me why `__typeof__(val) __val = (val)` is needed since `lf_schedule_with_value` takes a `void*`. In fact, why is `lf_schedule_with_val` needed?  It seems kind of risky, in fact, because it hides the fact that a pointer is passed to `lf_schedule_with_value`. Users are likely to use `lf_schedule_with_val` with the value argument being on the stack, which will not work.

Reply:
small additional note the `lf_schedule_with_val` and `lf_schedule_without_val` are for internal use only (see the `@private` in-front) they are used by the `lf_schedule` macro, which is variadic  (see the `LF_SCHEDULE_CHOOSER` magic)